internal/generator/iop/template/polynomial.go.tmpl

```diff
@@ -101,69 +94,31 @@ func (p *Polynomial) SetSize(size int) {
 	p.size = size
 }
 
-// SetBlindedSize sets the blinded size of the polynomial.
-func (p *Polynomial) SetBlindedSize(size int) {
-	p.blindedSize = size
-}
-
-// Blind blinds a polynomial q by adding Q(X)*(X^{n}-1),
-// where deg Q = blindingOrder and Q is random, and n is the
-// size of q. Sets the result to p and returns it.
-//
-// blindingOrder is the degree of Q, where the blinding is Q(X)*(X^{n}-1)
-// where n is the size of p. The size of p is modified since the underlying
-// polynomial is of bigger degree now. The new size is p.Size+1+blindingOrder.
-//
-// /!\ The code panics if wq is not in canonical, regular layout
-func (p *Polynomial) Blind(blindingOrder int) *Polynomial {
-	// check that p is in canonical basis
-	if p.Form != canonicalRegular {
-		panic("the input must be in canonical basis, regular layout")
-	}
-
-	// we add Q*(x^{n}-1) so the new size is deg(Q)+n+1
-	// where n is the size of wq.
-	newSize := p.size + blindingOrder + 1
-
-	// Resize p. The size of wq might has already been increased
-	// (e.g. when the polynomial is evaluated on a larger domain),
-	// if that's the case we don't resize the polynomial.
-	p.grow(newSize)
-
-	// blinding: we add Q(X)(X^{n}-1) to P, where deg(Q)=blindingOrder
-	var r fr.Element
-
-	for i := 0; i <= blindingOrder; i++ {
-		r.SetRandom()
-		(*p.coefficients)[i].Sub(&(*p.coefficients)[i], &r)
-		(*p.coefficients)[i+p.size].Add(&(*p.coefficients)[i+p.size], &r)
-	}
-	p.blindedSize = newSize
-
-	return p
-}
-
 // Evaluate evaluates p at x.
 // The code panics if the function is not in canonical form.
 func (p *Polynomial) Evaluate(x fr.Element) fr.Element {
 
+	if p.Basis == LagrangeCoset {
```

Comment:
question; seems p.Basis is only used in the iop package. Do we need it exposed? 

Wondering if this should be private to avoid accidental changes in the basis by copies or whatever, that could lead to a state where p.Basis is wrongly set and the x.Div(x, p.Coset) is a div by 0 that fails silently.